### DISCUSSION TITLE
[FEATURE REQUEST]: Mocking support

### DISCUSSION THREAD

**Discussion by thrixton** [dotnet/spark#469] [Q&A] [answered]
Is it absolutely necessary to seal all classes. e.g. SparkSession?
This is problematic for mocking as Moq (and others) cannot mock sealed classes.

**Comment by imback82** [comment, marked as answer]
They are sealed because we don't expect them to be extended. Can't you wrap them with your interfaces?

**Comment by thrixton** [comment]
Thanks @imback82,
I can wrap them but that then requires extra maintenance and I'm no longer using the original interface so lose intellisense documentation etc...

An alternative would be to extract an interface and then I could mock the interface.
I'd be happy to do the work if it would help and a PR would be pulled.
Thoughts?

It's fine if you don't want to, I just thought I'd check :)

Thanks for your help.
John
